### PR TITLE
Add -disableRedirects flag to disable following of redirects

### LINKED ISSUE #40: Add a flag to disable following of redirects
I have a use-case where I'm wanting to test the performance of an endpoint that is expected to return a 302. In the local testing that I've done, hey seems to silently follow the redirects via the `net/http` default behavior.

It might be handy to add a flag to disable the following of redirects for situations like this. I'd be happy to add one if you think it might be useful to others.

### PR DESCRIPTION
**Pull request by mark-adams**
```
hey -disable-redirects https://avatar-cdn.atlassian.com/8cfadb857fea2ad383ec7b7d5c88c5cf
154 requests done.
All requests done.

Summary:
  Total:	0.6952 secs
  Slowest:	0.3705 secs
  Fastest:	0.0137 secs
  Average:	0.1313 secs
  Requests/sec:	287.6955

Status code distribution:
  [302]	200 responses

Response time histogram:
  0.014 [1]	|∎
  0.049 [65]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.085 [51]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.121 [5]	|∎∎∎
  0.156 [0]	|
  0.192 [26]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.228 [1]	|∎
  0.263 [7]	|∎∎∎∎
  0.299 [3]	|∎∎
  0.335 [31]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.370 [10]	|∎∎∎∎∎∎

Latency distribution:
  10% in 0.0225 secs
  25% in 0.0373 secs
  50% in 0.0648 secs
  75% in 0.2460 secs
  90% in 0.3203 secs
  95% in 0.3354 secs
  99% in 0.3591 secs
```

Fixes #40

### CODE REVIEW

**Comment by mark-adams** [comment]
When you have a moment to review @rakyll, I'd appreciate it. 😄

**Comment by mark-adams** [comment]
Hi @rakyll,

It's been a few months since this was submitted. If you have a moment for a review I'd appreciate it.

Have a good one!

File: requester/requester.go [hunk 1]
```diff
@@ -222,7 +232,15 @@ func (b *Work) runWorker(n int) {
 	} else {
 		tr.TLSNextProto = make(map[string]func(string, *tls.Conn) http.RoundTripper)
 	}
-	client := &http.Client{Transport: tr, Timeout: time.Duration(b.Timeout) * time.Second}
+
+	var checkRedirect func(req *http.Request, via []*http.Request) error
+	if b.DisableRedirects {
+		checkRedirect = func(req *http.Request, via []*http.Request) error {
+			return errRedirectNotAllowed
```
Comment (rakyll) [inline requester/requester.go]: return http.ErrUseLastResponse instead. http.Client knows what http.ErrUseLastResponse is.

File: requester/requester.go [hunk 2]
```diff
@@ -222,7 +232,15 @@ func (b *Work) runWorker(n int) {
...
+		}
+	}
+
+	client := &http.Client{Transport: tr, Timeout: time.Duration(b.Timeout) * time.Second, CheckRedirect: checkRedirect}
```
Comment (rakyll) [inline requester/requester.go]: keep this line as it is.

then add 

if b.DisableRedirects {
   client.CheckRedirect = func (req *http.Request, via []*http.Request) error { ... } 
}

File: requester/requester.go (hunk 1 above)
Comment (mark-adams) [inline requester/requester.go]: :+1: I didn't know about `http.ErrUseLastResposne`. That's awesome!

**Comment by mark-adams** [comment]
@rakyll I've updated the PR with the changes requested. Thanks for the review!

**Comment by rakyll** [comment]
@mark-adams, thanks!

**Pull request merged**